packages/framework/src/Model/Product/ProductDataFactory.php

```diff
@@ -183,6 +186,9 @@ protected function fillFromProduct(ProductData $productData, Product $product):
         $productData->sellingTo = $product->getSellingTo();
         $productData->sellingDenied = $product->isSellingDenied();
         $productData->unit = $product->getUnit();
+        $productData->promotionXyData = $product->getPromotionXy() === null ?
+            $this->productPromotionXyDataFactory->create() :
+            $this->productPromotionXyDataFactory->createFromEntity($product->getPromotionXy());
```

Comment:
when product has no promotion, why set the data to empty promotion? shouldn't be this fact represented by null value?